### PR TITLE
ENG 2129: Cleans up workflow header

### PR DESCRIPTION
**Pull request by vsreekanti**
## Describe your changes and why you are making these changes

This PR cleans up the workflow header by:
* Removing the title and the old status chit from the workflow header.
* Adding the status icon to the breadcrumbs.
* Using the new `EngineItem` component to show the engine rather than displaying plaintext.
* Adding an icon for the next workflow run.
* Adding a version label to the version selector. 

## Related issue number (if any)

ENG-2129

## Loom demo (if any)
image

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Review by eunice-chan — APPROVED** [review]
Looks good!

File: src/ui/common/src/components/layouts/NavBar.tsx [hunk 1]
```diff
@@ -92,33 +98,37 @@ const NavBar: React.FC<{
...
-              >
-                {link.name}
-              </Link>
-            );
-          })}
-        </Breadcrumbs>
+            })}
+          </Breadcrumbs>
+
+          {afterBreadcrumbAdornment}
```
[the remark this answers is not in the document]
Comment (vsreekanti) [reply src/ui/common/src/components/layouts/NavBar.tsx:130]: Interesting idea. I'm kinda torn on this one — I see what you're saying, Eunice, but I also feel like afterwards is more natural. @agiron123, any thoughts on this?

File: src/ui/common/src/components/workflows/version_selector.tsx [hunk 2]
```diff
@@ -35,31 +29,7 @@ export const VersionSelector: React.FC = () => {
...
-        case ExecutionStatus.Failed:
-          menuItemIcon = (
-            <Box sx={{ fontSize: '20px', color: theme.palette.red['500'] }}>
-              <FontAwesomeIcon icon={faCircleXmark} />
-            </Box>
-          );
-          break;
-      }
-
+      const menuItemIcon = <StatusIndicator status={r.status} />;
```
[the remark this answers is not in the document]
Comment (vsreekanti) [reply src/ui/common/src/components/workflows/version_selector.tsx:32]: Andre added this relatively recently. It'll be good to use this everywhere, so we can unify the iconography and color choices.

File: src/ui/common/src/components/layouts/NavBar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (agiron123) [reply src/ui/common/src/components/layouts/NavBar.tsx:130]: As a general rule in most cases, icons are used to communicate a label, so they go on the left side. Some cases, (really just execution status and check status, we put them on the left).

I know it's not 100% consistent, but that's what we've been doing so far.

As for the adornment, I wouldn't like to see it in the nav breadcrumbs personally, but something we can do here is put the status icon on the left as Eunice suggested and have an 8px marginRight.

**Comment by cw75** [comment]
@vsreekanti Is this PR still relevant?

**Comment by cw75** [comment]
@vsreekanti ^^

**Comment by vsreekanti** [comment]
Sorry, missed your previous ping. Yes, we need to do some more work here, and I want to keep this around to finalize the design first. I'm hoping to have that done this week and will close this out afterwards.

**Comment by vsreekanti** [comment]
Closing this — subsumed by #1045.

**Pull request closed without merging**